pkg/sources/adapter/azureservicebussource/adapter.go

```diff
@@ -230,22 +248,49 @@ func connectionStringFromEnvironment(namespace, entityPath string) string {
 //  Both (DataAction):
 //  - Microsoft.ServiceBus/namespaces/messages/receive/action
 func (a *adapter) Start(ctx context.Context) error {
+	maxMessages := 5
 	logging.FromContext(ctx).Info("Listening for messages")
 
-	handle := a.msgRcvr.Listen(ctx, servicebus.HandlerFunc(a.handleMessage))
-	<-handle.Done()
-	return handle.Err()
+	messages, err := a.msgRcvr.ReceiveMessages(ctx, maxMessages, nil)
```

Comment:
If we don't run this in a loop (cancelable by context like in e.g. the source for Amazon DynamoDB), won't the adapter terminate immediately after receiving and processing _one_ batch of messages?